### PR TITLE
Added navbar.

### PR DESCRIPTION
**Pull request by handasolo**
Thanks to @ManrajGrover for fixing the bug that was giving me sleepless nights :D . P.S. @CuriousLearner Don't know if I did good with git flow or not, please check.

### CODE REVIEW

File: src/client/js/script.js [hunk 1]
```diff
@@ -63,4 +64,601 @@ $(document).ready(function() {
...
+
+    $('#nav-main .toggle').addClass('open');
+
+    // add click handler and set submenu class on submenus
+    NavMain.mainMenuLinks
+        .addClass('submenu-item')
+        .click(NavMain.handleSubmenuClick);
+
+    // focus first item
+    //$('#nav-main-menu [tabindex=0]').get(0).focus();
```
Comment (manrajgrover) [inline src/client/js/script.js]: You can remove this if commented and not needed.

File: src/client/js/script.js [hunk 2]
```diff
@@ -63,4 +64,601 @@ $(document).ready(function() {
...
+      // Open the menu
+      new_li.addClass('hover').find('[aria-expanded="false"]').attr('aria-expanded', 'true');
+      if (prev_li) {
+        // Close the last selected menu
+        prev_li.dequeue();
+      }
+    } else {
+      prev_li.clearQueue();
+    }
+  })/*bind('mouseout focusout', function(event) {
```
Comment (manrajgrover) [inline src/client/js/script.js]: Same here. You can remove the commented portion.

File: src/client/css/style.css [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/* CSS for all */
+	/* CSS for all */
```
Comment (manrajgrover) [inline src/client/css/style.css]: Nit. Extra space.

File: src/client/js/script.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (manrajgrover) [inline src/client/js/script.js]: Not sure why is this here.

File: src/client/js/script.js [hunk 5]
```diff
@@ -63,4 +64,601 @@ $(document).ready(function() {
...
+NavMain.init = function()
+{
+    NavMain.mainMenuItems = $('#nav-main .has-submenus > li');
+    NavMain.mainMenuLinks = $('#nav-main ul > li > [tabindex="0"]');
+
+    NavMain.mainMenuItems
+        .bind('mouseover focusin', NavMain.handleFocusIn)
+        .bind('mouseout focusout', NavMain.handleFocusOut)
+        .each(NavMain.initSubmenu);
+    console.log("Initializing..");
```
Comment (manrajgrover) [inline src/client/js/script.js]: I forgot to remove some `console.log()`. Sorry. Please remove them now.

File: src/client/js/script.js [hunk 6]
```diff
@@ -63,4 +64,601 @@ $(document).ready(function() {
...
+
+NavMain.handleFocusIn = function(e)
+{
+    var item = $(this);
+
+    if (NavMain.previousMenuItem) {
+        if (NavMain.previousMenuItem.attr('id') !== item.attr('id')) {
+            // Close the last selected menu
+            NavMain.previousMenuItem.dequeue();
+            console.log("Focus In dequeue..");
```
Comment (manrajgrover) [inline src/client/js/script.js]: Nit. Another `console.log()`.

File: src/client/js/script.js [hunk 7]
```diff
@@ -63,4 +64,601 @@ $(document).ready(function() {
...
+        } else {
+            NavMain.previousMenuItem.clearQueue();
+            console.log("Focus In empty..");
```
Comment (manrajgrover) [inline src/client/js/script.js]: Nit. Another `console.log()`.

**Comment by manrajgrover** [comment]
@Slayslot Reviewed your PR. Please make the required changes. I haven't gone through the logic part for now. Most of them are on additional portion that is not required which can be removed. Apart from that please follow 2 space indention rule as major part of code is already formatted in that way. This will ensure uniformity. :blush:

File: src/client/js/script.js (hunk 4 above)
Comment (CuriousLearner) [inline src/client/js/script.js]: Because Merge was not made right.
This indicates where was your head when the merge conflict happened.

**Comment by CuriousLearner** [comment]
Please resolve the merge conflicts properly. You can use MergeTool rather than manually doing it.

File: src/client/js/script.js (hunk 4 above)
Comment (handasolo) [inline src/client/js/script.js]: Yes, I thought I removed them all.

File: src/client/js/script.js (hunk 2 above)
Comment (handasolo) [inline src/client/js/script.js]: Did you make this comment and the one at 475 on your end? I can't remember doing this.

**Comment by handasolo** [comment]
@ManrajGrover Totally forgot about the 2 space indentation. @CuriousLearner Yes.

**Comment by anupkumarmishra** [comment]
@CuriousLearner  Please merge pull request if you find it okay .

File: src/client/js/script.js (hunk 2 above)
Comment (manrajgrover) [inline src/client/js/script.js]: Yeah. There was no point of having a `focusout` event because we want to close it on click.

Comment (handasolo) [inline src/client/js/script.js]: Alright, will remove.

**Comment by handasolo** [comment]
@ManrajGrover Please, do the honours.

File: src/client/js/script.js [hunk 8]
```diff
@@ -62,5 +62,582 @@ $(document).ready(function() {
...
+        }
+        if (target) {
+          target.get(0).focus(); // Sometimes target.focus() doesn't work
+          return false;
+        }
+        return true;
+      });
+    });
+  });
+
```
Comment (manrajgrover) [inline src/client/js/script.js:643]: @Slayslot It is throwing error here. Please check your code if its working or not because it isn't.

File: src/client/js/script.js [hunk 9]
```diff
@@ -62,5 +62,582 @@ $(document).ready(function() {
     $('#name,#message,#email').css('border-color','#484848');
     ctr = 0;
     validate_form();
-  }); 
+  });
+/*
```
Comment (manrajgrover) [inline src/client/js/script.js:67]: This is the cause of the error. You have deleted the closing brackets here. Please make the change.

**Comment by manrajgrover** [comment]
@Slayslot Reviewed your code. The feature isn't working. I have found the errors and commented. Please make the change.

**Comment by handasolo** [comment]
Forgot to add a title to this sorry. Fixed the problem. Please check.

**Comment by manrajgrover** [comment]
Looks good now. This can be merged now @anupkumarmishra . :blush:
